Engine/Entities/Ants/Ant.cs

```diff
@@ -100,23 +100,16 @@ public Ant(string name, Transform transform, World world) : this(name, transform
         }
 
         //TODO: Could be added to a higher level of entity. The only problem is that it depends on PerceptionDistance so maybe in LivingEntity?
-        
+
         /// <summary>
         /// Generates a list of the entities that are in this Ant's perceptionDistance. 
         /// </summary>
         /// <returns>List of the entities in the perception range of this Ant</returns>
-        public List<Entity> GetSurroundingEntities<T>() where T : Entity
+        public List<T> GetSurroundingEntities<T>() where T : Entity
         {
-            List<Entity> list = new();
-            
-            foreach (Entity e in World.Entities)
-            {
-                if (e is not T) continue;
-                if (!(e.Transform.GetDistance(Transform) <= PerceptionDistance)) continue;
-                list.Add(e);
-            }
+            int radius = Math.Max((int) MathF.Ceiling(PerceptionDistance / World.WorldDivision), 1);
 
-            return list;
+            return World.CheckEntitiesInRegion<T>(Region.X, Region.Y, radius);
```

Comment:
This does not check for exact perception distance anymore, it only gets the region but an entity could be in an overlapping region and not in the perception field

Reply:
Fixed